src/utils/queryFacade.utils.ts

```diff
@@ -0,0 +1,146 @@
+import { Query } from '@utils/types.utils';
+import { Vulnerability } from '@utils/types.utils';
+import { VulDatabase } from '@utils/types.utils';
+import axios from 'axios';
+import e from 'express';
+
+/***************** GET VULNERABILITIES FROM EXT DATABASES ****************************/
+export async function getVulnerabilities(query: Query) {
+  let config: any = {
+    method: query.method,
+    url: query.url,
+    headers: {
+      [query.headers.authKey]: query.headers.authValue,
+    },
+    params: {
+      [query.params.searchKey]: query.params.searchValue,
+    },
+    data: query.body,
+  };
+
+  return axios(config)
+    .then((response) => {
+      return response.data;
+    })
+    .catch(function (error) {
+      if (error.response) {
+        // The request was made and the server responded with a status code
+        // that falls out of the range of 2xx
+        console.log(error.response.data);
+        console.log(error.response.status);
+        console.log(error.response.headers);
+      } else if (error.request) {
+        // The request was made but no response was received
+        // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
+        // http.ClientRequest in node.js
+        console.log(error.request);
+      } else {
+        // Something happened in setting up the request that triggered an Error
+        console.log('Error', error.message);
+      }
+      console.log(error.config);
+      return error.response.status;
+    });
+}
+
+/****************** RESPONSE CLEANER INTERFACE **********************/
+interface QueryCleaner {
+  (rawResponse: any): Vulnerability[];
+}
+
+//Cleaner instances
+let nvdCleaner: QueryCleaner;
+let sonatypeCleaner: QueryCleaner;
+
+//Cleaner implementaions
+nvdCleaner = function (rawResponse): Vulnerability[] {
+  let vulns: Vulnerability[] = [];
+  let rawVulns: any = rawResponse.vulnerabilities; //get all vulnerabilities
+
+  //Create Vulnerability for each cve in response
+  rawVulns.forEach(function (cve: any) {
+    let v: Vulnerability = {
+      cveId: cve['cve'].id,
+      packgaeId: -1,
+      impact: -1,
+      likelihood: -1,
+      risk: -1,
+      cvss2: '',
+    };
+    //Get correct CVSS version
+    if (cve['cve']['metrics'].hasOwnProperty('cvssMetricV2')) {
+      //V2
+      v.cvss2 =
+        cve['cve']['metrics']['cvssMetricV2'][0]['cvssData']['vectorString'];
+    } else if (cve['cve']['metrics'].hasOwnProperty('cvssMetricV31')) {
```

Comment:
if we're planning not to have cvss v3 in maybe we should get rid of this. but if we're planning to map v3 -> v3, we should probably fill in this statement before merging.

Reply:
What do you think about opening an issue for this?
I was thinking cvss mapping would be included as a Vulnerability analysis utility (and be importable)

Reply:
thats a good idea!